pkg/util/workloadrepo/worker_test.go

```diff
@@ -132,15 +229,567 @@ func TestGlobalWorker(t *testing.T) {
+		first := true
+		for _, p := range partitions {
+			if !first {
+				fmt.Fprint(sb, ", ")
+			}
+			fmt.Fprint(sb, buildPartitonRow(p))
+			first = false
+		}
+		fmt.Fprint(sb, `)`)
+	}
+	return sb.String()
+}
+
+func createTableWithParts(ctx context.Context, t *testing.T, tk *testkit.TestKit, tbl *repositoryTable,
+	sess sessionctx.Context, partitions []time.Time) {
+	createSQL, err := buildCreateQuery(ctx, sess, tbl)
+	require.NoError(t, err)
+	createSQL += buildPartitionString(partitions)
+	tk.MustExec(createSQL)
+	require.True(t, validatePartitionsMatchExpected(ctx, t, sess, tbl, partitions))
+}
+
+func validatePartitionCreation(ctx context.Context, now time.Time, t *testing.T,
+	sess sessionctx.Context, tk *testkit.TestKit,
+	firstTestFails bool, tableName string, partitions []time.Time, expectedParts []time.Time,
+) {
+	tbl := getTable(t, tableName)
+	createTableWithParts(ctx, t, tk, tbl, sess, partitions)
+
+	is := sess.GetDomainInfoSchema().(infoschema.InfoSchema)
+	require.False(t, firstTestFails == checkTableExistsByIS(ctx, is, tbl.destTable, now))
+
+	is = sess.GetDomainInfoSchema().(infoschema.InfoSchema)
+	require.NoError(t, createPartition(ctx, is, tbl, sess, now))
+
+	require.True(t, validatePartitionsMatchExpected(ctx, t, sess, tbl, expectedParts))
+
+	is = sess.GetDomainInfoSchema().(infoschema.InfoSchema)
+	require.True(t, checkTableExistsByIS(ctx, is, tbl.destTable, now))
+}
+
+func TestCreatePartition(t *testing.T) {
+	ctx, store, dom, addr := setupDomainAndContext(t)
+	wrk := setupWorker(ctx, t, addr, dom, "worker", false)
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("create database WORKLOAD_SCHEMA")
+	tk.MustExec("use WORKLOAD_SCHEMA")
+
+	_sessctx := wrk.getSessionWithRetry()
+	sess := _sessctx.(sessionctx.Context)
+
+	fillInTableNames()
+
+	now := time.Now()
+
+	/* Tables without partitions are not currently supported.  This also does
+	   not test cases with partitions in the future beyond what will be created. */
+
+	// Should create one partition for today and tomorrow on a table with only old partitions before today.
+	partitions := []time.Time{now.AddDate(0, 0, -1)}
+	expectedParts := []time.Time{now.AddDate(0, 0, -1), now, now.AddDate(0, 0, 1)}
+	validatePartitionCreation(ctx, now, t, sess, tk, true, "PROCESSLIST", partitions, expectedParts)
+
+	// Should create one partition for tomorrow on a table with only a partition for today.
+	partitions = []time.Time{now}
+	expectedParts = []time.Time{now, now.AddDate(0, 0, 1)}
+	validatePartitionCreation(ctx, now, t, sess, tk, true, "DATA_LOCK_WAITS", partitions, expectedParts)
+
+	// Should not create any partitions on a table with only a partition for tomorrow.
+	partitions = []time.Time{now.AddDate(0, 0, 1)}
+	expectedParts = []time.Time{now.AddDate(0, 0, 1)}
+	validatePartitionCreation(ctx, now, t, sess, tk, false, "TIDB_TRX", partitions, expectedParts)
+
+	// Should not create any partitions on a table with only a partitions for both today and tomorrow.
```

Comment:
Nit: typo (should be "with only partitions for")

Reply:
Fixed